staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -87,12 +87,18 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 			return
 		}
 
+		// send first event for non-read-only requests
+		ri, _ := request.RequestInfoFrom(ctx)
+		if !(ri != nil && ri.ReadOnly) {
```

Comment:
My feeling is that we should make this decision in a single place, and I'd like an option to disable this stage. We talked about putting the option in the audit backend, but if we're already deciding whether to create a RequestReceived event here, perhaps this would be a good place to check that option as well?